### ISSUE TITLE
Compatibility issue with com.amplifyframework:aws-storage-s3 2.14.8 library due to signature changes in WorkManager 2.9.0

### ISSUE THREAD

**Issue by geek5nan** [aws-amplify/amplify-android#2669]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin - Coroutines

### Amplify Categories

Storage

### Gradle script dependencies

<details>

```groovy
// Put output below this line
implementation("androidx.work:work-runtime-ktx:2.9.0")
implementation("com.amplifyframework:aws-core:2.14.8")
implementation("com.amplifyframework:aws-storage-s3:2.14.8")
implementation("com.amplifyframework:aws-auth-cognito:2.14.8")

```

</details>

### Environment information

<details>

```
# Put output below this line
------------------------------------------------------------
Gradle 7.5
------------------------------------------------------------

Build time:   2022-07-14 12:48:15 UTC
Revision:     c7db7b958189ad2b0c1472b6fe663e6d654a5103

Kotlin:       1.6.21
Groovy:       3.0.10
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          17.0.9 (JetBrains s.r.o. 17.0.9+0-17.0.9b1087.7-11185874)
OS:           Mac OS X 14.0 aarch64


```

</details>

### Describe the bug

I've noticed an incompatibility issue when using WorkManager 2.9.0 in my application with `com.amplifyframework:aws-storage-s3:2.14.8` library.

Specifically, I found that WorkManager 2.9.0 modified the signature of the onStop function in androidx.work.ListenableWorker: from stop()V to stop(I)V. This has led to unexpected behavior in the RouterWoker.onStopped function within the com.amplifyframework:aws-storage-s3:2.14.8 library, triggering a NoSuchMethodError.

Prior to this, my application was working fine with earlier versions of WorkManager.
```
Fatal Exception: java.lang.NoSuchMethodError: No virtual method stop()V in class Landroidx/work/c; or its super classes (declaration of 'androidx.work.c' appears in base.apk)
       at com.amplifyframework.storage.s3.transfer.worker.RouterWorker.onStopped(RouterWorker.kt:93)
       at androidx.work.ListenableWorker.stop(ListenableWorker.java:294)
       at androidx.work.impl.WorkerWrapper.interrupt(WorkerWrapper.java:386)
       at androidx.work.impl.Processor.interrupt(Processor.java:440)
       at androidx.work.impl.Processor.stopWork(Processor.java:260)
       at androidx.work.impl.utils.StopWorkRunnable.run(StopWorkRunnable.kt:51)
       at androidx.work.impl.utils.SerialExecutorImpl$Task.run(SerialExecutorImpl.java:96)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
       at java.lang.Thread.run(Thread.java:1012)
```

### Code Snippet

```java
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by tylerjroach** [comment]
@geek5nan Thank you for the report. We will look at bumping the version and look into different API usages to prevent this from happening in the future.

**Comment by tylerjroach** [comment]
@geek5nan Please upgrade to Amplify Android 2.14.9. This should resolve the issue for you.

**Comment by geek5nan** [comment]
@tylerjroach Got it, thank you for your quick response. I will upgrade to 2.14.9 as soon as possible and observe the improvements.

**Comment by mattcreaser** [comment]
This issue was resolved in PR #2674. If any problems persist please open a new issue, thanks!
